File: app/schemas/internship.py
from datetime import date, datetime

from pydantic import BaseModel, Field

from app.models.enums import InternshipStatus


class InternshipCreateRequest(BaseModel):
    title: str = Field(..., max_length=255)
    description: str | None = None
    city: str | None = Field(default=None, max_length=100)
    direction: str | None = Field(default=None, max_length=100)
    salary: int | None = None
    deadline: date | None = None


class InternshipUpdateRequest(BaseModel):
    title: str | None = Field(default=None, max_length=255)
    description: str | None = None
    city: str | None = Field(default=None, max_length=100)
    direction: str | None = Field(default=None, max_length=100)
    salary: int | None = None
    deadline: date | None = None
    status: InternshipStatus | None = None


class InternshipResponse(BaseModel):
    id: int
    employer_id: int
    title: str | None = None
    description: str | None = None
    city: str | None = None
    direction: str | None = None
    salary: int | None = None
    status: InternshipStatus | None = None
    deadline: date | None = None
    created_at: datetime | None = None


class InternshipPublicResponse(BaseModel):
    id: int
    title: str | None = None
    description: str | None = None
    city: str | None = None
    direction: str | None = None
    salary: int | None = None
    status: InternshipStatus | None = None
    deadline: date | None = None
    created_at: datetime | None = None
    company_name: str | None = None

File: frontend/src/types/internship.ts
export type InternshipStatus = "ACTIVE" | "CLOSED" | "ARCHIVED";

/**
 * Соответствует `InternshipPublicResponse` на бэкенде:
 * `GET /api/v1/internships/{internship_id}` и `GET /api/v1/internships`.
 */
export type InternshipPublicResponse = {
  id: number;
  title: string | null;
  description: string | null;
  city: string | null;
  direction: string | null;
  salary: number | null;
  status: InternshipStatus | null;
  deadline: string | null; // JSON date => строка (например "2026-03-23")
  created_at: string | null; // JSON timestamp => строка
  company_name: string | null;
};

